internal/extsvc/types.go

```diff
@@ -113,3 +116,116 @@ func WebhookURL(kind string, externalServiceID int64, externalURL string) (strin
 	// eg. https://example.com/.api/github-webhooks?externalServiceID=1
 	return fmt.Sprintf("%s/.api/%s?%s=%d", externalURL, path, IDParam, externalServiceID), nil
 }
+
+// Common describes basic fields expected to exist on an external service
+// TODO: This is a hack needed because we have ExternalService defined in three places
+// cmd/repo-updater/repos
+// cmd/frontend/types
+// internal/api
+// Until we simplify this, RateLimits below will ask for just the fields it needs
+type Common struct {
+	Config      string
+	Kind        string
+	DisplayName string
+}
+
+// RateLimits returns rate limit config for the supplied external services
+func RateLimits(services []Common) ([]RateLimitConfig, error) {
```

Comment:
Does this need to take a slice? If you change it to

```suggestion
func ExtractRateLimitConfig(displayName, kind, config string) (RateLimitConfig, error) {
```

and move the loops to the callers, you can get rid of the whole `Common` thing which I don't think we should introduce.

Reply:
Good idea, I initially wanted to encapsulate the logic in the `extsvc` package, including the check if the err is a `ErrRateLimitUnsupported`. I think I can still do that by returning a bool indicating if rate limiting is supported for the host.

Reply:
Actually no, that gets confusing. I'll just document it.